kratos/tests/test_redistance.py

```diff
@@ -47,8 +47,8 @@ def test_model_part_sub_model_parts(self):
             max_distance = max(max_distance, d)
             min_distance = min(min_distance, d)
 
-        self.assertAlmostEqual(max_distance, 0.44556526310761013)
-        self.assertAlmostEqual(min_distance,-0.504972246827639)
+        self.assertAlmostEqual(max_distance, 0.5011280499206787)

```

Comment:
Why is this here? Was the test failing?

Reply:
yeah, I had to add Conditions Blocks to mdpas. Afterwards it started to fail. I checked, existing DISTANCE calculator doesn't use NORMAL variable, or any normal_calculation_utils methods. So this should be because of change in mdpas.

As I see, VariationalDistanceCalculation Process is run only with 2 max iterations, which is not enough for convergent solution. So I increased number of max iterations untill convegent solution is given. Once that solution is obtained, I used that value. Do you also have some tests which uses VariationalDistanceCalculationProcess? If so could you please also check.

Reply:
I checked with current master as well, If you increase `max_iterations`, then It will again fail. But it will pass with increased amount of iterations with the value I have provided in this PR.

https://github.com/KratosMultiphysics/Kratos/blob/1d5ddbd3b218c8e7e3cb49566ede6d4898fd3453/kratos/tests/test_redistance.py#L40-L51

Reply:
I agree that 2 are too few iterations but I wouldn't change it here. If test fails with max_iterations=2, something is wrong somewhere.

Reply:
The difference is coming because, when we have conditions block in mdpa, it marks nodes belonging to conditions with `BOUNDARY` = `true`. And in `DistanceCalculationElementSimplex`, it adds additional `RHS` contribution when `BOUNDARY` of a node is true as in following line.

https://github.com/KratosMultiphysics/Kratos/blob/1d5ddbd3b218c8e7e3cb49566ede6d4898fd3453/kratos/elements/distance_calculation_element_simplex.h#L260-L264

This is the reason for change in reference values. This difference in reference values can be eliminated by having a new PR to increase `max_iterations` and then getting the reference values from that to pass this test since at the convergence those values are the same.

Reply:
After discussing with @philbucher , I reverted changes done to existing mdpas and `test_redistance.py` and introduced `coarse_spehere_with_conditions.mdpa` to be used in `test_normal_utils.py`.

Reply:
that makes sense to me. Thanks!

Reply:
What I do is using SkinDetectionProcess instead of creating a new mdpa

Reply:
> After discussing with @philbucher , I reverted changes done to existing mdpas and `test_redistance.py` and introduced `coarse_spehere_with_conditions.mdpa` to be used in `test_normal_utils.py`.

I was using SkinDetectionProcess, why did you change it?

Reply:
Although usually I would say that it is better to re-use things I wold say that in this case it makes sense to have another file that is self contained and we know that it works

Using the skin-detection process adds another layer of complexity to an already existing complex case
Plus then we need to write and re-read the mdpa file that the process generates which is also not an optimal solution

maybe once the skin detection also works in MPI we could come back to this discussion and think if we want to use it, but for the moment I would not

Reply:
Yes, it is something want to do in the sort term